src/super_gradients/common/crash_handler/crash_tips.py

```diff
@@ -0,0 +1,209 @@
+
+
+class CrashTip:
+    """Base class to add tips to exceptions raised while using SuperGradients.
+
+    A tip is a more informative message with some suggestions for possible solutions or places to debug.
+    """
+
+    @classmethod
+    def is_relevant(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> bool:
+        """
+        Check if this tip is relevant.
+
+        Beside the class, the input params are as returned by sys.exc_info():
+            :param cls:             Class inheriting from CrashTip
+            :param exc_type:        Type of exception
+            :param exc_value:       Exception
+            :param exc_traceback:   Traceback
+
+            :return:                True if the current class can help with the exception
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def _get_explanation(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        """
+        Propose an explanation on what caused the exception to be raised.
+
+        Beside the class, the input params are as returned by sys.exc_info():
+            :param cls:             Class inheriting from CrashTip
+            :param exc_type:        Type of exception
+            :param exc_value:       Exception
+            :param exc_traceback:   Traceback
+
+            :return:                Explanation
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def _get_solution(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        """
+        Propose a solution (or more) to avoid the exception
+
+        Beside the class, the input params are as returned by sys.exc_info():
+            :param cls:             Class inheriting from CrashTip
+            :param exc_type:        Type of exception
+            :param exc_value:       Exception
+            :param exc_traceback:   Traceback
+
+            :return:                Solution
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def _get_tip(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        """
+        Provide a customized tip for the exception, combining explanation and solution.
+
+        Beside the class, the input params are as returned by sys.exc_info():
+            :param cls:             Class inheriting from CrashTip
+            :param exc_type:        Type of exception
+            :param exc_value:       Exception
+            :param exc_traceback:   Traceback
+
+            :return:                Tip
+        """
+        explanation = indent_string(cls._get_explanation(exc_type, exc_value, exc_traceback), indent_size=4)
+        solution = indent_string(cls._get_solution(exc_type, exc_value, exc_traceback), indent_size=4)
+
+        msg = (
+            "- What went wrong:\n"
+            f"{explanation}\n"
+            "- How to solve it:\n"
+            f"{solution}\n"
```

Comment:
Question:
Is it better to force the split of "Problem explanation" / "Solution" with 2 functions _get_explanation, _get_solution.
OR is it better to give the freedom to organise the tip in any way ?

Reply:
yes. it should be `_get_explanation() -> str` and `_get_solutions() -> List(str)`

Reply:
In 8f1cc5d28b908456ea4accb8e65e4a66a267abbc I merged explanation and solution into _get_tips() -> List[str]